units/src/timestamp.rs

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! A UNIX timestamp used as the Bitcoin block time.
+//!
+//! Also known as Epoch Time - January 1, 1970.
+//!
+//! This differs from other UNIX timestamps in that we only use non-negative values. The Epoch
+//! pre-dates Bitcoin so timestamps before this are not useful for block timestamps.
+
+use core::ops;
+
+#[cfg(feature = "arbitrary")]
+use arbitrary::{Arbitrary, Unstructured};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// A Bitcoin block timestamp.
+///
+/// > Each block contains a Unix time timestamp. In addition to serving as a source of variation for
+/// > the block hash, they also make it more difficult for an adversary to manipulate the block chain.
```

Comment:
nit: could probably drop `the block chain` from the end of the sentence without loss of clarity.